openhtf/output/callbacks/__init__.py

```diff
@@ -122,3 +126,9 @@ def __call__(self, test_record):
       else:
         raise TypeError('Expected string or iterable but got {}.'.format(
             type(serialized_record)))
+
+  @staticmethod
+  def _get_valid_filename(s):
+      # Reference: https://stackoverflow.com/a/46801075
+      s = str(s).strip().replace(' ', '_')
+      return re.sub(r'(?u)[^-\w.]', '_', s)
```

Comment:
I've hit an issue before where the filename was too long for the filesystem, this could be guarded against too here.